RecoEgamma/EgammaTools/src/GBRForestTools.cc

```diff
@@ -27,13 +27,34 @@ namespace {
 
 };
 
-std::unique_ptr<const GBRForest> GBRForestTools::createGBRForest(const std::string &weightFile){
-  edm::FileInPath weightFileEdm(weightFile);
-  return GBRForestTools::createGBRForest(weightFileEdm);
+// Creates a pointer to new GBRForest corresponding to a TMVA weights file,
+// given a file path as string.  When the file is not found, also the same name
+// appended with ".gz" and ".gzip" is tried.  This allows the maintainters of
+// the data repositories to gzip weight files a posteriori without changing
+// anything else.
+std::unique_ptr<const GBRForest> GBRForestTools::createGBRForest(const std::string &weightFile,
+                                                                 std::vector<std::string> &varNames){
+
+   std::string gzipExtensions[3] = {"gz", "gzip"};
```

Comment:
Hello, I didn't find a way to avoid the exception handling completely, but now I catch only the edm exceptions with the FileInPathError category.

Reply:
why ```[3]``` ?

Reply:
Excuse me, should be [2]. In the beginning I had one more possible ending in mind and that is a remnant of it.